packages/dds/tree/src/codec/codec.ts

```diff
@@ -44,6 +46,45 @@ export interface SchemaValidationFunction<Schema extends TSchema> {
 	check(data: unknown): data is Static<Schema>;
 }
 
+/**
+ * A kind of validator for SharedTree's internal data formats.
+ * @remarks
+ * Assuming not data corruption or type confusion, such validation should never fail due to careful format versioning.
```

Comment:
```suggestion
 * Assuming no data corruption or type confusion, such validation should never fail when used alongside careful format versioning.
```

Reply:
This is a user facing comment, and the data versioning is something we do internally not something users are expected to do. From a user perspective, unless the pass the wrong data to some API, validation should be a no op since compatibility checking is done via version checks which are not part of the validator. Thus I disagree with your phrasing.

Reply:
I have updated this to try and be more clear.